lib/wibox/init.lua

```diff
@@ -347,10 +356,16 @@ local function new(args)
 
     -- If a value is not found, look in the drawin
     setmetatable(ret, {
-        __index = w,
+        __index = function(self, k)
+            if wibox["get_"..k] then
+                return self["get_"..k](self)
+            else
+                return w[k]
+            end
+        end,
         __newindex = function(self, k,v)
-            if wibox["set_"..k] then
-                wibox["set_"..k](v)
+            if rawget(self, "set_"..k) then
+                self["set_"..k](self, v)
```

Comment:
> (Why does this need/use rawget?)

To avoid getting handled by the metatable in an infinite loop. Eventually, it will hit the drawin __index, then the drawin miss handler. This is not what we need. That would prevent wibox polymorphism as used by wibars, tooltips, notifications and menus.